backend/test/til_web/controllers/activities/reaction_controller_test.exs

```diff
@@ -0,0 +1,174 @@
+defmodule TilWeb.Activities.ReactionControllerTest do
+  use TilWeb.ConnCase
+  import Til.Guardian
+  import Til.Factory
+  alias Til.ShareableContent
+  alias Til.Repo
+  alias Til.Activities.Reaction
+
+  describe "POST /api/posts/reactions" do
+    test "creates reaction given type for post when not reacted yet and returns ok", %{conn: conn} do
+      current_user = insert(:user)
+      {:ok, token, _} = encode_and_sign(current_user.uuid, %{})
+      type = "like"
+
+      post = insert(:post)
+      insert(:reaction, user_id: current_user.id, post_id: post.id, type: "love")
+
+      response =
+        conn
+        |> put_req_header("authorization", "bearer: " <> token)
+        |> post(Routes.post_reaction_path(conn, :react, post.id), %{
+          type: type
+        })
+
+      assert response.status == 200
+
+      reactions = Repo.all(Reaction)
+
+      assert length(reactions) == 2
+
+      [_, reaction] = reactions
+
+      assert reaction.user_id == current_user.id
+      assert reaction.post_id == post.id
+      assert reaction.type == type
+    end
+
+    test "does not create new reaction and throws error when post is already reacted with given type", %{conn: conn} do
+      current_user = insert(:user)
+      {:ok, token, _} = encode_and_sign(current_user.uuid, %{})
+      type = "like"
+
+      post = insert(:post)
+      insert(:reaction, user_id: current_user.id, post_id: post.id, type: type)
+
+      response =
+        conn
+        |> put_req_header("authorization", "bearer: " <> token)
+        |> post(Routes.post_reaction_path(conn, :react, post.id), %{
+          type: type
+        })
+
+      assert response.status == 400
+
+      reactions = Repo.all(Reaction)
+
+      assert length(reactions) == 1
+
+      {:ok, parsed_response_body} = Jason.decode(response.resp_body)
+
+      assert parsed_response_body["errors"] == %{"post_id" => ["has already been taken"]}
```

Comment:
it's a matter of taste but i prefer to do empty lines when i group code in logical chunks, for example:
```ex

assert response.status == 400
{:ok, parsed_response_body} = Jason.decode(response.resp_body)
assert parsed_response_body["errors"] == %{"post_id" => ["has already been taken"]}

reactions = Repo.all(Reaction)
assert length(reactions) == 1
```
so first chunk is response related, other db related, in that case you don't need to do so much spacing,
in your code, you have the lines in the following order:
- response related (`assert response.status == 400`)
- db related (`reactions = Repo.all(Reaction)`)
- db related (` assert length(reactions) == 1`)
- response related (`{:ok, parsed_response_body} = Jason.decode(response.resp_body)`)
- response related (`assert parsed_response_body["errors"] == %{"post_id" => ["has already been taken"]}`)

I assume this might have happened because of TDD but i think it's ok to rearrange that later. Of course you will find situations where that's impossible to do.

Consider this as a hint, not a requirement to change the code.